### DISCUSSION TITLE
skip windowless space when cycling spaces

### DISCUSSION THREAD

**Discussion by lilruffian** [asmvik/yabai#1003] [Q&A]
is it possible to skip over windowless spaces when activating `yabai -m space —focus next`?

I have 8 mission control spaces set up in my workspace. Some apps are rule assigned to certain spaces. At times a space will be blank if it’s rule assigned app window is deactivated. I’d like to skip past blanks when cycling through spaces. Is this an option already baked in to yabai? If not, has someone already solved for this with a script  ? If so, please share! Thanks!

**Comment by Inc0n** [comment]
Perhaps you can do a windows count in the space like 
```
yabai -m query --windows --space 2
```
Then parse the output with `jq`.
From there you can determine whether to skip this space or not.
